apps/openmw/mwworld/containerstore.hpp

```diff
@@ -68,9 +69,9 @@ namespace MWWorld
             MWWorld::CellRefList<ESM::Repair>            repairs;
             MWWorld::CellRefList<ESM::Weapon>            weapons;
 
-            std::map<std::string, int> mLevelledItemMap;
-            ///< Stores result of levelled item spawns. <refId, count>
-            /// This is used to remove the spawned item(s) if the levelled item is restocked.
+            std::map<std::string, std::pair<int, std::string> > mLevelledItemMap;
```

Comment:
This data structure does not correctly handle the case where the same item ID is selected by multiple levelled lists.

E.g.

levi_1 can contain item_a, item_b. RNG selects item_a
levi_2 can contain item_a, item_c. RNG selects item_a

With this map we can only store one link i.e. to levi_1 or levi_2, but we need to store both.